src/postcard_generator.py

```diff
@@ -0,0 +1,59 @@
+import speech_recognition as sr
+import streamlit as st
+import torch
+from diffusers import StableDiffusionPipeline
+from PIL import Image
+
+MODEL_ID = "runwayml/stable-diffusion-v1-5"
+IMAGES_DIR = "../images/"
+SYSTEM = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+@st.cache(allow_output_mutation=True, suppress_st_warning=True)
+def get_pipe():
+    if SYSTEM == "cuda":
+        return StableDiffusionPipeline.from_pretrained(MODEL_ID, torch_dtype=torch.float16, revision="fp16").to(SYSTEM)
+    else:
+        return StableDiffusionPipeline.from_pretrained(MODEL_ID).to(SYSTEM)
+
+
+def model_image(txt, theme, steps, height, width):
+    pipe = get_pipe()
+    pipe.safety_checker = lambda images, clip_input: (images, False)
+    prompt = f"Cartoon {theme} postcard {txt}"
+    image = pipe(prompt=prompt, num_inference_steps=steps, height=height, width=width).images[0]
+    return image
+
+
+def record_voice():
+    try:
+        listener = sr.Recognizer()
+        with sr.Microphone() as microphone:
+            aud_msg = listener.listen(microphone)
+        txt_msg = listener.recognize_google(aud_msg, language='en-US')
+    except Exception as e:
+        st.sidebar.exception(e)
+        st.sidebar.write("We gave you back kitty ʕ ᵔᴥᵔ ʔ")
+        txt_msg = "kitty"
+    return txt_msg.lower()
```

Comment:
What if I say `kitty`? The app would handle this as an error?
You need to handle exception more accurately, the most simple solution is to return `None`.

Also, showing the user an exception message is a bad idea. In most cases, the user understands nothing. You should handle errors by yourself and show a beautiful message that describes what went wrong and how to fix it.

Reply:
Done!